### PR TITLE
UCT: Remove MAX_SHORT attribute from UCT config

### PR DESCRIPTION
**Pull request by dmitrygladkov**
## What

Removes `MAX_SHORT` attribute from UCT config

## Why ?

Only TCP and MM transports depend on this attribute. They can be replaced by the transport-related attributes (`FIFO_ELEM_SIZE` for MM)

## How ?

1. Removes max_short from config
2. Remove dependencies from CUT transports on this parameter
3. Remove excessive UD_MIN_INLINE macro
4. Remove unnecessary test that checks MAX_SHORT config

### CODE REVIEW

**Comment by dmitrygladkov** [comment]
@brminich 
@alinask this restores the FIFO element size parameter (instead of MAX_SHORT) added in #386
@MattBBaker there are changes in ugni
can you review pls?

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9934/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9935/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
> Test FAILed.
> See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9935/ for details (Mellanox internal link).

build timed out

**Comment by dmitrygladkov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9940/ for details (Mellanox internal link).

File: src/uct/ugni/rdma/ugni_rdma_iface.c [hunk 1]
```diff
@@ -12,7 +12,7 @@
 
 static ucs_config_field_t uct_ugni_rdma_iface_config_table[] = {
     /* This tuning controls the allocation priorities for bouncing buffers */
-    { "", "MAX_SHORT=2048;MAX_BCOPY=2048;ALLOC=huge,mmap,heap", NULL,
+    { "", "ALLOC=huge,mmap,heap", NULL,
```
Comment (brminich) [inline src/uct/ugni/rdma/ugni_rdma_iface.c]: why did you remove `MAX_BCOPY` also?

File: test/gtest/uct/test_p2p_am.cc [hunk 2]
```diff
@@ -551,7 +551,8 @@ const unsigned uct_p2p_am_misc::RX_QUEUE_LEN = 64;
 
 UCS_TEST_SKIP_COND_P(uct_p2p_am_misc, no_rx_buffs, RUNNING_ON_VALGRIND) {
 
-    mapped_buffer sendbuf(10 * sizeof(uint64_t), SEED1, sender());
+    mapped_buffer sendbuf(ucs_min(sender().iface_attr().cap.am.max_short,
```
Comment (brminich) [inline test/gtest/uct/test_p2p_am.cc]: why need this change?

File: src/uct/sm/mm/base/mm_iface.c [hunk 3]
```diff
@@ -43,9 +43,13 @@ static ucs_config_field_t uct_mm_iface_config_table[] = {
      "Possible values are:\n"
      " y   - Allocate memory using huge pages only.\n"
      " n   - Allocate memory using regular pages only.\n"
-     " try - Try to allocate memory using huge pages and if it fails, allocate regular pages.\n",
+     " try - Try to allocate memory using huge pages and if it fails, allocate regular pages.",
      ucs_offsetof(uct_mm_iface_config_t, hugetlb_mode), UCS_CONFIG_TYPE_TERNARY},
 
+    {"FIFO_ELEM_SIZE", "128",
+     "Size of the FIFO element size (dtat + header) in the MM UCTs.",
```
[the remark this answers is not in the document]
Comment (dmitrygladkov) [reply src/uct/sm/mm/base/mm_iface.c]: 👍

File: src/uct/ugni/rdma/ugni_rdma_iface.c (hunk 1 above)
Comment (dmitrygladkov) [reply src/uct/ugni/rdma/ugni_rdma_iface.c]: UGNI doesn't use `MAX_SHORT` and `MAX_BCOPY` config attributes

File: test/gtest/uct/test_p2p_am.cc (hunk 2 above)
Comment (brminich) [reply test/gtest/uct/test_p2p_am.cc]: why is to fit into short required here?

File: src/uct/ugni/rdma/ugni_rdma_iface.c (hunk 1 above)
Comment (brminich) [reply src/uct/ugni/rdma/ugni_rdma_iface.c]: ok

File: test/gtest/uct/test_p2p_am.cc (hunk 2 above)
Comment (dmitrygladkov) [reply test/gtest/uct/test_p2p_am.cc]: > why is to fit into short required here?

It will fail w/ the following error:
`UCX  ERROR Invalid am_short length: 80 (expected: <= 64)`

Comment (brminich) [reply test/gtest/uct/test_p2p_am.cc]: ic, it uses uct am_short for sending

**Comment by alinask** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9957/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
> Test FAILed.
> See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9957/ for details (Mellanox internal link).

related to #3514

**Comment by dmitrygladkov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9960/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9966/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9967/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9970/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
bot:mlx:retest

**Comment by dmitrygladkov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9975/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9980/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9984/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/9988/ for details (Mellanox internal link).

**Comment by dmitrygladkov** [comment]
@yosefe ok to merge?
